include/mbedtls/pem.h

```diff
@@ -103,6 +108,23 @@ int mbedtls_pem_read_buffer( mbedtls_pem_context *ctx, const char *header, const
                      const unsigned char *pwd,
                      size_t pwdlen, size_t *use_len );
 
```

Comment:
In Mbed TLS 2.x, you had to figure out the lifetime of `ctx->buf` yourself. By reading the library's source code, you could deduce that `ctx->buf` is allocated with `mbedtls_calloc`, and that you could either:

* keep using `ctx->buf`, then call `mbedtls_pem_free`;
* or make a copy of the pointer `ctx->buf`, set `ctx->buf = 0`, call `mebdtls_pem_free(ctx)`, and keep using the buffer for a while then call `mbedtls_free` on it;
* or make a copy of the pointer `ctx->buf`, set `ctx->buf = 0`, call `mebdtls_pem_read_buffer(ctx)` again, and keep using the buffer for a while then call `mbedtls_free` on it.

We're now ruling out the second and third workflow. I don't necessarily mind, but I'd like to make it more explicit in the documentation. In particular, the documentation of `mbedtls_pem_read_buffer` should mention that you must not call this function twice on the same buffer without an intervening call to `mbedtls_pem_free`.

Reply:
I attempted to update the comment in the header to add the above warning/expectation.